portfolio/src/main/webapp/script.js

```diff
@@ -12,26 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Identity theft is not a joke', 'I declare bankruptcy', 'I am a little stitious'];
+getComments();
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
+// Submits a comment to be stored in the server.
+document.querySelector('.comment-form').addEventListener('submit', async function(e) {
+  e.preventDefault();
+  const response = await fetch('/comment', {
+    method: 'POST',
+    body: new URLSearchParams(new FormData(event.target)),
+  });
+  const json = await response.json();
+  if ('error' in json) {
+    console.log('Could not submit comment');
+  }
+  const commentContainer = document.querySelector('.comments');
+  let liComment = document.createElement('li');
+  liComment.innerText = document.querySelector('.comment-box').value;
+  liComment.className = 'comment';
+  commentContainer.append(liComment);
+});
 
 async function getComments() {
-  const response = await fetch('/data');
+  const response = await fetch('/comment');
   const json = await response.json();
-  const commentContainer = document.getElementById('comments');
+  const commentContainer = document.querySelector('.comments');
+  let liComment;
   for (const comment of json.comments) {
-    commentContainer.innerHTML += comment;
+    liComment = document.createElement('li');
```

Comment:
since you repeat this block of code above (line 28), this would be cleaner if you created a new function:

```
function appendComment(value) {
  const commentContainer = document.querySelector('.comments');
  let liComment = document.createElement('li');
  liComment.innerText = value;
  liComment.className = 'comment';
  commentContainer.append(liComment);
}
```

That way if you make changes to this in the future, you only have to update it in one place.